gem install error: Undefined symbols for architecture arm64

### What version of gRPC and what language are you using?
grpc: 1.56.0
language: ruby 3.2.2

### What operating system (Linux, Windows,...) and version?
macOS 13.4
Apple M1 chip

### What runtime / compiler are you using (e.g. python version or version of gcc)
Apple clang version 14.0.3 (clang-1403.0.22.14.1)

### What did you do?
`gem install grpc -v 1.56.0`

### What did you expect to see?
Gem successfully installed.

### What did you see instead?
`
40 warnings generated.
compiling rb_enable_cpp.cc
linking shared-object grpc/grpc_c.bundle
Undefined symbols for architecture arm64:
  "_CFRelease", referenced from:
      absl::lts_20230125::time_internal::cctz::local_time_zone() in libgrpc.a(time_zone_lookup.o)
  "_CFStringGetCString", referenced from:
      absl::lts_20230125::time_internal::cctz::local_time_zone() in libgrpc.a(time_zone_lookup.o)
  "_CFStringGetLength", referenced from:
      absl::lts_20230125::time_internal::cctz::local_time_zone() in libgrpc.a(time_zone_lookup.o)
  "_CFStringGetMaximumSizeForEncoding", referenced from:
      absl::lts_20230125::time_internal::cctz::local_time_zone() in libgrpc.a(time_zone_lookup.o)
  "_CFTimeZoneCopyDefault", referenced from:
      absl::lts_20230125::time_internal::cctz::local_time_zone() in libgrpc.a(time_zone_lookup.o)
  "_CFTimeZoneGetName", referenced from:
      absl::lts_20230125::time_internal::cctz::local_time_zone() in libgrpc.a(time_zone_lookup.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [grpc_c.bundle] Error 1

make failed, exit code 2
`

--- a/src/ruby/ext/grpc/extconf.rb
+++ b/src/ruby/ext/grpc/extconf.rb
@@ -141,7 +141,10 @@ end
 
 ext_export_file = File.join(grpc_root, 'src', 'ruby', 'ext', 'grpc', ext_export_filename())
 $LDFLAGS << ' -Wl,--version-script="' + ext_export_file + '.gcc"' if linux
-$LDFLAGS << ' -Wl,-exported_symbols_list,"' + ext_export_file + '.clang"' if apple_toolchain
+if apple_toolchain
+  $LDFLAGS << ' -weak_framework CoreFoundation' if RUBY_PLATFORM =~ /arm64/
+  $LDFLAGS << ' -Wl,-exported_symbols_list,"' + ext_export_file + '.clang"'
+end
 
 $LDFLAGS << ' ' + File.join(grpc_lib_dir, 'libgrpc.a') unless windows
 if grpc_config == 'gcov'

